src/Data/Vec.agda

```diff
@@ -159,6 +161,13 @@ foldl₁ _⊕_ (x ∷ xs) = foldl _ _⊕_ x xs
 sum : ∀ {n} → Vec ℕ n → ℕ
 sum = foldr _ _+_ 0
 
+count : ∀ {a p} {A : Set a} {P : Pred A p} → Decidable P →
```

Comment:
I would prefer this to accept a function `A -> Bool` rather than a decidable predicate. Not everything you want to count will necessarily have a decidability proof.

Reply:
A boolean version is immediately recoverable, no?
```agda
countBool : ∀ {a} {A : Set a} → (A → Bool) → ∀ {n} → Vec A n → ℕ
countBool f = count (λ x → f x ≟ true)
```

Reply:
True, but that argument goes the other way too, and I think there's a lot to be said for the boolean version's simpler type.

Reply:
You and your obsession with bools (and my obsession with predicates) :stuck_out_tongue: Well idiomatic Agda is to encode properties as predicates rather than assigning true or false to every instance, so I'd argue that most people should be using predicates by default anyway. Plus we've just changed `filter`, `break`, `takeWhile` etc. in `Data.List` to use predicates over bools because they were much more difficult to use and prove properties with bools. I feel we should be consistent.

Reply:
> Plus we've just changed filter, break, takeWhile etc. in Data.List to use predicates over bools because they were much more difficult to use and prove properties with bools.

This is disappointing news! I've personally never found a situation where a bool makes things more difficult to prove, whereas decidable predicates add noise that makes it harder for me to get my head around what Agda is expecting from me in some situations! However, if this is the way the standard library is going then I guess it is best to stick to consistency, so fine, I'll stop arguing that stance.

Reply:
Sorry if you're working in a context that using `bool`s is fine, I realise the changes are probably quite annoying. We have left in the boolean versions however, although they are deprecated.

However most of the library is based around predicates: `All`, `Any`, `_∈ xs`, `_≤ 7`, `_| 3`, `Prime` etc. and there are no equivalent boolean functions, so I think it probably makes sense to optimise for the most common use case.